lightning/src/ln/features.rs

```diff
@@ -692,25 +757,35 @@ impl<T: sealed::ShutdownAnySegwit> Features<T> {
 		self
 	}
 }
-
-impl<T: sealed::Context> Writeable for Features<T> {
-	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
-		(self.flags.len() as u16).write(w)?;
-		for f in self.flags.iter().rev() { // Swap back to big-endian
-			f.write(w)?;
+macro_rules! impl_feature_len_prefixed_write {
+	($features: ident) => {
+		impl Writeable for $features {
+			fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+				(self.flags.len() as u16).write(w)?;
+				self.write_be(w)
+			}
+		}
+		impl Readable for $features {
+			fn read<R: io::Read>(r: &mut R) -> Result<Self, DecodeError> {
+				Ok(Self::from_be_bytes(Vec::<u8>::read(r)?))
+			}
 		}
-		Ok(())
 	}
 }
+impl_feature_len_prefixed_write!(InitFeatures);
+impl_feature_len_prefixed_write!(ChannelFeatures);
+impl_feature_len_prefixed_write!(NodeFeatures);
+impl_feature_len_prefixed_write!(InvoiceFeatures);
 
-impl<T: sealed::Context> Readable for Features<T> {
+impl Writeable for ChannelTypeFeatures {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+		self.write_be(w)
```

Comment:
Why not just use `impl_feature_len_prefixed_write` for simplicity? If not, comment would be good